tests/ui/codegen/subtyping-enforces-type-equality.rs

```diff
@@ -1,5 +1,6 @@
-// run-pass
+// check-pass
```

Comment:
maybe let's do `build-pass`? Does this not get triggered by this code -- I would expect a stderr file here.